Alignment/TrackerAlignment/plugins/CosmicRateAnalyzer.cc

```diff
@@ -0,0 +1,529 @@
+
+      virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+      virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+      static double stampToReal(edm::Timestamp time) {
+	return time.unixTime() + time.microsecondOffset()*1e-6;
+      }
+      void ClearInEventLoop();
+      void ClearInEndRun();
+      // ----------member data ---------------------------
+      edm::EDGetTokenT<reco::TrackCollection> trackTags_;
+      edm::EDGetTokenT<edmNew::DetSetVector<SiStripCluster> > clustercollectionToken_;
+      edm::EDGetTokenT<reco::MuonCollection> muonTags_;
+      edm::RunNumber_t lastrunnum;
+      double lastruntime;
+      edm::Service<TFileService> fs;
+   
+      unsigned int DetectorID; 
+      TTree* treeEvent;
+      TTree* treeRun;
+      TTree* treeCluster;
+
+      int			events;
+      int       		track_BPIX  ; 	
+      int       		track_FPIX  ; 	
+      int       		track_PIXEL ; 	
+      int       		track_TEC   ; 	
+      int       		track_TECM  ; 	
+      int       		track_TECP  ; 	
+      int       		track_TOB   ; 	
+      int       		track_TIB   ;     
+      int       		track_TID   ;     
+      int       		track_TIDM  ;     
+      int       		track_TIDP  ; 	
+      
+      std::vector<int>		v_ntrk;
+      int 			ntrk;
+      int			ntrk_runnum;
+   
+      int			number_of_tracks;
+      int			number_of_tracks_PIX;
+      int			number_of_tracks_FPIX;
+      int			number_of_tracks_BPIX;
+      int			number_of_tracks_TEC;
+      int			number_of_tracks_TECP;
+      int			number_of_tracks_TECM;
+      int			number_of_tracks_TOB;
+      int			number_of_tracks_TIB;
+      int			number_of_tracks_TID;
+      int			number_of_tracks_TIDP;
+      int			number_of_tracks_TIDM;
+      int			number_of_events;
+      edm::RunNumber_t		runnum;
+      double                    run_time ;
+      std::vector<double>	pt;
+      std::vector<double>	charge;
+      std::vector<double>	chi2;
+      std::vector<double>	chi2_ndof;
+      std::vector<double>	eta;
+      std::vector<double>	theta;
+      std::vector<double>	phi;
+      std::vector<double>	p;
+      std::vector<double>	d0;
+      std::vector<double>	dz;
+      std::vector<double>	nvh;
+      std::vector<double>	DTtime;
+
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+CosmicRateAnalyzer::CosmicRateAnalyzer(const edm::ParameterSet& iConfig):
+  trackTags_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("tracks"))),
+  clustercollectionToken_(consumes<edmNew::DetSetVector<SiStripCluster> >(iConfig.getParameter<edm::InputTag>("tracks"))),
+  muonTags_(consumes<reco::MuonCollection>(iConfig.getParameter<edm::InputTag>("muons")))
+{
+   //now do what ever initialization is needed
+   //
+    treeEvent = fs->make<TTree>("Event","");
+    treeRun = fs->make<TTree>("Run","");
+    treeCluster = fs->make<TTree>("Cluster","");
+
+}
```

Comment:
@skeshri: you are using the TFileService, i.e. you have to add `usesResource(TFileService::kSharedResource);` in the constructor
in addition you need to add the template parameter `edm::one::SharedResources` to the `edm::one::EDAnalyzer` from which you inherit